### ISSUE TITLE
:edit/file needs :new before :bdel/bwipe

### ISSUE THREAD

**Issue by lfmet** [junegunn/vader.vim#134]
[1] Not ok (the results gets copied to the test file):
```vader
Before:
  edit A
After:
  bdelete A
Execute:
  Log bufnr('A')
Execute:
  Log bufnr('A')
```
[2] Ok
```vader
Execute:
  edit A
Execute:
  Log bufnr('A')
Execute:
  bdelete A

Execute:
  edit A
Execute:
  Log bufnr('A')
Execute:
  bdelete A
```

Isn't [1] supposed to be executed like [2]?

**Comment by blueyed** [comment]
AFAIK "After" gets executed after all Execute blocks only.
It is useful when used in a single test file / suite.

**Comment by blueyed** [comment]
The documentation could be improved in that regard maybe: https://github.com/junegunn/vader.vim#syntax-of-vader-file

..and assuming that I am understanding it myself correctly.

**Comment by lfmet** [comment]
```vader
Before:
  Log 'before'
After:
  Log 'after'
Execute:
Execute:
```

Gives
```
> before
(1/2) [EXECUTE]
> after
> before
(2/2) [EXECUTE]
> after
```

**Comment by blueyed** [comment]
ok, that's good after all.

Then it might be an issue with what the current buffer is on `After`?!

Can you provide a failing test for Vader itself?

**Comment by lfmet** [comment]
The problem is with `bdelete`, after that, it doesn't go back to `[Vader-workbench]`.
```vader
Execute:
  " [Vader-workbench]
  Log bufname('')
  edit A

  " A
  Log bufname('')
  bdelete A

  " [Vader]
  Log bufname('')

" Execute (the [Vader] output will be here):
```

**Comment by blueyed** [comment]
Yeah, as a workaround I am using `new` before any `edit` or `file` in tests.

**Comment by blueyed** [comment]
@junegunn 
Do you think this could be addressed somehow without a `new´?

**Comment by junegunn** [comment]
I think we could set up a buffer-local `BufDelete` autocmd during the execution. When the user accidentally (or intentionally) deletes the buffer, it can either recreate the buffer, or warn the user and suspend the test immediately.

**Comment by blueyed** [comment]
Sounds like a good idea!

@lfmet 
Would you like to fix this?

**Comment by lfmet** [comment]
I'm not sure how to handle this:
```vader
Execute:
  " [Vader-workbench]
  edit A
  " A
  bdel A
  " [Vader]
Execute (output pasted here, [Vader-workbench] doesn't exist anymore):
```
This can be solved with something like: ` if tabpagenr() != s:workbench_tab | tabnew | endif` in [here](https://github.com/junegunn/vader.vim/blob/master/autoload/vader/window.vim#L37).
Or add a `BufDelete` in the [lines](https://github.com/junegunn/vader.vim/blob/master/autoload/vader/window.vim#L77) to recreate the tab.

A `BufDelete` on  `[Vader-workbench]` would solve this (?)(should be solved?):
```vader
Execute:
  bdel
Execute:
```
